src/Servers/Kestrel/Core/src/KestrelConfigurationLoader.cs

```diff
@@ -435,14 +436,98 @@ private X509Certificate2 LoadCertificate(CertificateConfig certInfo, string endp
             }
             else if (certInfo.IsFileCert)
             {
-                var env = Options.ApplicationServices.GetRequiredService<IHostEnvironment>();
-                return new X509Certificate2(Path.Combine(env.ContentRootPath, certInfo.Path), certInfo.Password);
+                var environment = Options.ApplicationServices.GetRequiredService<IHostEnvironment>();
+                if (certInfo.KeyPath != null)
```

Comment:
Can we not infer from file extension here? Seems a bit weird to move pfx till last if we know how to handle pfx.

Reply:
We don't infer from the file extension here.

If you give us Path, only we assume it is a pfx. If you give us Path and KeyPath we assume it is a cert + key in PEM format.